src/providers/BackgroundBlurProvider/index.tsx

```diff
@@ -88,7 +93,11 @@ const BackgroundBlurProvider: FC<Props> = ({ spec, options, children }) => {
   );
 
   useEffect(() => {
-    initializeBackgroundBlur();
+    initializeBackgroundBlur().then((createdProcessor) => {
+      if (observer) {
+        (createdProcessor as BackgroundBlurProcessor).addObserver(observer);
```

Comment:
Do we have to do this? I think the `useEffect` will run based on `processor` state update and should add the observer. Could you verify once?

Also suggest adding log statements if that helps in debugging.

Reply:
Yeah, thanks for catching that. Now that we have the processor in that useEffect, we wont need this.